pallets/loans/src/lib.rs

```diff
@@ -1051,11 +1057,33 @@ pub mod pallet {
 				.map_err(|_| Error::<T>::NoLoanChangeId.into())
 		}
 
-		fn update_portfolio_valuation_for_pool(
+		fn registered_prices(
+			pool_id: T::PoolId,
+		) -> Result<BTreeMap<T::PriceId, T::Balance>, DispatchError> {
+			let collection = T::PriceRegistry::collection(&pool_id)?;
+			Ok(ActiveLoans::<T>::get(pool_id)
+				.iter()
+				.filter_map(|(_, loan)| loan.price_id())
+				.filter_map(|price_id| {
+					collection
+						.get(&price_id)
+						.map(|price| (price_id, price.0))
+						.ok()
+				})
+				.collect::<BTreeMap<_, _>>())
+		}
+
+		pub fn update_portfolio_valuation_for_pool(
 			pool_id: T::PoolId,
+			input_prices: PriceCollectionInput<T>,
 		) -> Result<(T::Balance, u32), DispatchError> {
 			let rates = T::InterestAccrual::rates();
-			let prices = T::PriceRegistry::collection(&pool_id)?;
+			let prices = match input_prices {
+				PriceCollectionInput::Empty => BTreeMap::default(),
+				PriceCollectionInput::Custom(prices) => prices.into(),
+				PriceCollectionInput::FromRegistry => Self::registered_prices(pool_id)?,
```

Comment:
What is the benfit of using the `PriceCollectionInput` here vs just using a `BTreeMap` as the parameter?

Reply:
You can see each use case here: https://github.com/centrifuge/centrifuge-chain/pull/1748/files#diff-efd52f7fa525763070479bd9d72c66286451a6752bed8534e83e1004e755c3e1R469

Reply:
`Empty` is just sugar for `Custom([])` and could be removed. But the caller should choose between `Custom` and `FromRegistry`.

Reply:
I prefer @lemunozm's proposal over just using a `BTreeMap` as it is more explicit which is always helpful for runtime API which can be expected to be used by externals as well.

Reply:
Okay, you changed the valuation parameters of the runtime-api. Sounds good!

 The most important thing though would be to mimic closing an epoch with 
* faked prices
* Optional<Solution>

Returning:
* Fullfillment amounts for each tranche for both invest and redeem
* New pool-reserve
* New Tranche prices

Reply:
Ok, I guess somehow this should be amalgamated with some other `pallet-pool` calls to get this, right?

Then I propose to create another in the PoolSystem RuntimeAPI, maybe called simulate_close_epoch or similar, where all these calls are performed to return all the required values above.

Reply:
Great, I'll do in another PR pool-related.